internal/generator/vector/output/cloudwatch/cloudwatch.go

```diff
@@ -0,0 +1,190 @@
+package cloudwatch
+
+import (
+	"fmt"
+	"strings"
+
+	logging "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	. "github.com/openshift/cluster-logging-operator/internal/generator"
+	. "github.com/openshift/cluster-logging-operator/internal/generator/vector/elements"
+	"github.com/openshift/cluster-logging-operator/internal/generator/vector/helpers"
+	vectorhelpers "github.com/openshift/cluster-logging-operator/internal/generator/vector/helpers"
+	"github.com/openshift/cluster-logging-operator/internal/generator/vector/output/security"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type CloudWatch struct {
+	ID_Key      string
+	Desc        string
+	ComponentID string
+	Inputs      string
+	Index       string
+	Endpoint    string
+	//if Endpoint provided then it overrides  Region         string
+	Host           string
+	GroupBy        string
+	LogGroupPrefix string
+
+	Cw_groupName  string
+	Cw_streamName string
+
+	SecurityConfig Element
+	EndpointConfig Element
+}
+
+func (e CloudWatch) Name() string {
+	return "cloudwatchTemplate"
+}
+
+func (e CloudWatch) Template() string {
+
+	return `{{define "` + e.Name() + `" -}}
+
+[sinks.{{.ComponentID}}]
+type = "aws_cloudwatch_logs"
+inputs = {{.Inputs}}
+endpoint = "{{.Endpoint}}"
+create_missing_group = true
+create_missing_stream = true
+compression = "none"
+encoding.codec = "json"
+request.concurrency = 2
+group_name  =  "{{"{{.Cw_groupName}}"}}"
+stream_name =  "{{"{{.Cw_streamName}}"}}"
+tls.verify_certificate = false
+{{- end}}
+`
+}
+
+func AddCwGroupNameCw_streamName(groupbytype string, logGroupPrefix string, id string, inputs []string) Element {
+	return Remap{
+		Desc:           "Adding group_name and stream_name field",
+		ComponentID:    id,
+		Inputs:         helpers.MakeInputs(inputs...),
+		GroupBy:        groupbytype,
+		LogGroupPrefix: logGroupPrefix,
+		VRL: strings.TrimSpace(`
+
+		Cw_groupName_app = "unknown"
+		Cw_streamName_app = "unknown"
+		Cw_groupName_infra = "unknown"
+		Cw_streamName_infra = "unknown"
+		Cw_groupName_audit = "unknown"
+		Cw_streamName_audit = "unknown"
+
+	if .GroupBy == "logType" {
+		Cw_groupName_app = "application"
+		Cw_streamName_app = "{{.ComponentID}}"
+		Cw_groupName_infra = "infrastructure"
+		Cw_streamName_infra = join!([{{.host}},".",{{.ComponentID}}])
```

Comment:
suggest not to use component id in the VRL. component ID is just used to name a source/sink/transform in vector config.